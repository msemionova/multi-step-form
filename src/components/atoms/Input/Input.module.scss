.input-field {
  .label {
    font: var(--typography-input-label);
    display: inline-block;
    margin-bottom: 1rem;
  }

  .input {
    font: var(--typography-input-text);
    background-color: var(--colore-white);
    border-radius: var(--border-radius-medium);
    border: var(--border-secondary);
    padding: 1.5rem;
    transition: border-color 200ms;
    width: 100%;
    color: var(--color-black-text);

    &:focus,
    &:active {
      outline: none;
      border-color: var(--color-primary-dark);
    }

    &::placeholder {
      color: var(--color-gray-text-light);
      opacity: 1;
    }

    &:-ms-input-placeholder {
      color: var(--color-gray-text-light);
    }

    &::-ms-input-placeholder {
      color: var(--color-gray-text-light);
    }
  }

  &:not(:last-child) {
    margin-bottom: 2.4rem;
  }
}
