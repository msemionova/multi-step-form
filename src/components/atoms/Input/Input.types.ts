export type InputProps = {
  id?: string,
  label?: string,
  type?: 'text' | 'tel' | 'password' | 'email' | 'hidden',
  placeholder?: string,
  pattern?: string,
  minlength?: number,
  maxlength?: number,
  name?: string,
  title?: string,
  value?: string,
  required?: boolean
}
