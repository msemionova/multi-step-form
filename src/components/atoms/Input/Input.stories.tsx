import type { Meta, StoryObj } from '@storybook/react';

import Input from './Input';

const meta: Meta<typeof Input> = {
  component: Input,
};

export default meta;
type Story = StoryObj<typeof Input>;

export const Default: Story = {
  args: {
    placeholder: 'Default input'
  }
};

export const Password: Story = {
  args: {
    placeholder: 'Password input',
    type: 'password'
  }
};

export const Email: Story = {
  args: {
    placeholder: 'Email input',
    type: 'email'
  }
};

export const WithLabel: Story = {
  args: {
    id: 'with_label',
    placeholder: 'Default input with label',
    label: 'Label for the input'
  }
};
