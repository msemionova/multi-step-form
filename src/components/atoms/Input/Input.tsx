import styles from './Input.module.scss';
import { InputProps } from './Input.types';

const Input = ({
  id,
  label,
  type = 'text',
  placeholder = '',
  pattern,
  minlength,
  maxlength,
  name = id,
  title = '',
  value,
  required
}: InputProps) => {
  const labelElement = label &&
    <label htmlFor={name} className={styles['label']}>{label}</label>;

  return (
    <div className={styles['input-field']}>
      {labelElement}
      <input
        className={styles['input']}
        id={id}
        type={type}
        placeholder={placeholder}
        pattern={pattern}
        minLength={minlength}
        maxLength={maxlength}
        name={name}
        title={title}
        value={value}
        required={required}
      />
    </div>
  );
};

export default Input;
