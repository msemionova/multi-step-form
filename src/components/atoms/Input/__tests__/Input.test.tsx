import Input from '../Input';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';

describe('<Input />', () => {
  it('renders correctly without required nor label...', () => {
    const tree = renderer
      .create(<Input />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
