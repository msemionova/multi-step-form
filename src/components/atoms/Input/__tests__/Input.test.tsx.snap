// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> renders correctly without required nor label... 1`] = `
<div
  className="input-field"
>
  <input
    className="input"
    placeholder=""
    title=""
    type="text"
  />
</div>
`;
